src/praisonai-agents/praisonaiagents/bots/run_status.py

```
@@ -0,0 +1,315 @@
+            except AttributeError:
+                supports = None
+        return bool(supports) if supports is not None else True
+
+    async def _render(self, *, emoji: str, label: str) -> None:
+        """Render the given emoji/label via the capability-gated callback."""
+        try:
+            if self._use_reactions() and self._set_reaction is not None:
+                await self._set_reaction(emoji)
+            elif self._edit_label is not None:
+                await self._edit_label(label)
+        except Exception as e:  # pragma: no cover - defensive; never break a run
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)